components/tidb_query_datatype/src/codec/datum.rs

```diff
@@ -235,6 +241,8 @@ impl Datum {
             Datum::Dur(ref d) => cmp_f64(d.to_secs_f64(), f),
             Datum::Time(t) => cmp_f64(t.convert(ctx)?, f),
             Datum::Json(_) => Ok(Ordering::Less),
+            Datum::Enum(_) => Ok(Ordering::Less),
```

Comment:
Is it possible to fully deprecate `Datum`?

Reply:
Let's give it a try.